_posts/2020-02-14-splunk-integration-with-ruby-on-rails.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title:  "Splunk Integration with Ruby on Rails (dokcer or not!)"
+date:   2020-01-10 10:11:00
+image:  
+tags:   [Splunk, Logging, Ruby, Rails, Docker]
+---
+
+This article is to walk you through setting up splunk and rails integration and not how to use splunk.
+
+**Stack and Versions:**
+- Docker: 19.03.5
+- Ruby: 2.6.4
+- Rails: 6.0
+- Gems:
+    + Required:
+        * rails_semantic_logger: 4.4 - 4.4.3
+    + Related:
+        * puma: 4.3.1
+- Splunk: 
+    + Locally hosted or Cloud (we will cover both)
+
+
+## Step 1: Splunk sign up
+[Create an account with splunk](https://www.splunk.com/), if you haven't one yet. You may choose to sign-up for the 14-day free trial for the cloud instance, or download splunk directly. We will cover both setups in this article!
+
+## Step 2: Install gems
+We're using `rails_sementic_logger` for logging and `puma` compiler
+
+## Step 3: Configure Splunk
+First you need to setup a token to use in your app or from the command line and this is how you do it:
+- Visit your cloud instance "https://prd-p-YOUR-INSTANCE.cloud.splunk.com" or your self-service setup "http://localhost:8000/"
+- Click on "Settings"
+- Under "DATA", choose "Data Inputs"
+- Under "Local inputs", choose "HTTP Event Collector"
+- Click on "New Token"
+- Give your token a name and follow steps till you create a token
+- Go back to "HTTP Event Collector" page and make sure the token in "Enabled"
+
+## Step 4: Use command-line to make sure your token is working
+Notice url is slightly different between the cloud instance and your local setup. We are looking for a success message: `{"text":"Success","code":0}`
+- Self-service cloud instances: https://input-prd-p-XXXXXXXX.cloud.splunk.com:8088
+- Managed cloud instances: https://http-inputs-XXXXXXXX.splunkcloud.com
+- Enterprise instances (also your local setup "localhost"): https://my-self-hosted-splunk.com:8088
+
+#### Self-service
+- From your local machine: `curl -k https://localhost:8088/services/collector/raw -H 'Authorization: Splunk YOUR_TOKEN' -d '{"event":"Hello, World!"}'`
```

Comment:
might be useful to explain the various ports splunk uses above in the intro to give more context for this